docs/ceph-csi-upgrade.md

```diff
@@ -65,6 +66,38 @@ controller and snapshot CRD. more info can be found
 Refer [upgrade-from-v3.2-v3.3](https://github.com/ceph/ceph-csi/blob/v3.3.1/docs/ceph-csi-upgrade.md)
 to upgrade from cephcsi v3.2 to v3.3
 
+## Upgrading from v3.4 to v3.5
+
+**Ceph-csi releases from devel are expressly unsupported.** It is strongly
+recommended that you use [official
+releases](https://github.com/ceph/ceph-csi/releases) of Ceph-csi. Unreleased
+versions from the devel branch are subject to changes and incompatibilities
+that will not be supported in the official releases. Builds from the devel
+branch can have functionality changed and even removed at any time without
+compatibility support and without prior notice.
+
+**Also, we do not recommend any direct upgrades to 3.5 except from 3.4 to 3.5.**
+For example, upgrading from 3.3 to 3.5 is not recommended.
+
+git checkout v3.5.0 tag
+
+```bash
+git clone https://github.com/ceph/ceph-csi.git
+cd ./ceph-csi
+git checkout v3.5.0
+```
+
+**Note:** While upgrading please Ignore warning messages from kubectl output
```

Comment:
change `we have successfully upgraded cephfs csi from v3.3 to v3.4` to 3.5 in multiple places